src/scanspec/service.py

```diff
@@ -186,40 +327,58 @@ def sub_sample(frames: Frames[str], ratio: float) -> Frames:
     return frames.extract(indexes, calculate_gap=False)
 
 
-def scanspec_schema_text() -> str:
-    """Return the text representation of the GraphQL schema."""
-    return graphql.utilities.print_schema(scanspec_schema)
+def _calc_smallest_step(points: List[np.ndarray]) -> float:
+    # Calc abs diffs of all axes, ignoring any zero values
+    absolute_diffs = [_abs_diffs(axis_midpoints) for axis_midpoints in points]
+    # Normalize and remove zeros
+    norm_diffs = np.linalg.norm(absolute_diffs, axis=0)
+    norm_diffs = norm_diffs[norm_diffs > 0.0]
+    # Return the smallest value (Aka. smallest step)
+    return np.amin(norm_diffs)
 
 
-def run_app(cors=False, port=8080):
-    """Run an application providing the scanspec service."""
-    app = web.Application()
+def _abs_diffs(array: np.ndarray) -> np.ndarray:
+    """Calculates the absolute differences between adjacent elements in the array.
 
-    view = GraphQLView(schema=scanspec_schema, graphiql=True)
+    Args:
+        array: A 1xN array of numerical values
 
-    # Make GraphQLView compatible with aiohttp-cors
-    # https://github.com/aio-libs/aiohttp-cors/issues/241#issuecomment-514278001
-    for method in ("GET", "POST", "PUT", "DELETE", "PATCH", "HEAD"):  # no OPTIONS
-        app.router.add_route(method, "/graphql", _asyncify(view), name="graphql")
+    Returns:
+        A newly constucted array of absolute differences
+    """
+    # [array[1] - array[0], array[2] - array[1], ...]
+    adjacent_diffs = array[1:] - array[:-1]
+    return np.absolute(adjacent_diffs)
 
-    # Optional, for adding batch query support (used in Apollo-Client)
-    # GraphQLView.attach(app, schema=schema, batch=True, route_path="/graphql/batch")
 
+def run_app(cors: bool = False, port: int = 8080) -> None:
+    """Run an application providing the scanspec service."""
     if cors:
-        # Configure default CORS settings.
-        cors_config = aiohttp_cors.setup(
-            app,
-            defaults={
-                "*": aiohttp_cors.ResourceOptions(
-                    allow_credentials=True,
-                    expose_headers="*",
-                    allow_headers="*",
-                )
-            },
+        app.add_middleware(
```

Comment:
Do we need CORS now? I only added it for graphiql...

Reply:
https://github.com/dls-controls/scanspec/issues/78